infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/ReplicationSession.java

```diff
@@ -43,25 +51,29 @@ public boolean equals(Object o) {
             return false;
         }
         ReplicationSession that = (ReplicationSession) o;
-        return remoteClusterId.equals(that.remoteClusterId) && subscriber.equals(that.subscriber);
+        return remoteClusterId.equals(that.remoteClusterId) && localClusterId.equals(that.localClusterId)
+                && subscriber.equals(that.subscriber);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(remoteClusterId, subscriber);
+        return Objects.hash(remoteClusterId, localClusterId, subscriber);
     }
 
     // TODO: To be removed after session is introduced in connections
```

Comment:
do we need this method now that the TODO is being addressed?

Reply:
My thought is that since ReplicationSession type is being removed in the SessionManager PR, it will be taken care of then. 

Also, places where ReplicationSession type is used will need to be replaced with the proto type. Since that would make this PR big, I'll take it up separately after the SessionManager PR.